### PR TITLE
Add support of NuoDB into querydsl-sql

### PR DESCRIPTION
**Pull request by tazija**
backport of https://github.com/querydsl/querydsl/pull/1267

### CODE REVIEW

**Comment by timowest** [comment]
@tazija Could you combine the commits?

File: querydsl-sql/src/main/java/com/mysema/query/sql/NuoDBTemplates.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        }
+    }
+
+    @Override
+    protected boolean requiresQuotes(String identifier, boolean precededByDot) {
+        return isReservedWord(identifier) || super.requiresQuotes(identifier, precededByDot);
+    }
+
+    private boolean isReservedWord(String identifier) {
+        return RESERVED_WORDS.contains(identifier.toUpperCase());
```
Comment (Shredder121) [inline querydsl-sql/src/main/java/com/mysema/query/sql/NuoDBTemplates.java]: I believe this code(line 169-175) is already in SQLTemplates.

**Comment by Shredder121** [comment]
Could you also revise the commit when you are done?
I *think* that [this change](https://github.com/querydsl/querydsl/pull/1273/files#diff-d24b83b49c360ad4512e7e44877ffd0bR1015) is maybe the result of a conflict not right resolved.

**Comment by timowest** [comment]
@tazija Can be reopened when the requested fixes are done.

**Pull request closed without merging**